### PR TITLE
fix(BurnTokensPopup): Handle non-integer values for assets 

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

- Support for non-integer values in `BurnTokensPopup`
- validation fixed
- Setting wrap to StatusRadioButton
- `AmountInput` - added possibility to customize err msg for exceeding max and label  and flag for zero as valid input

### Affected areas

`BurnTokensPopup`, `StatusRadioButton`, `AmountInput`

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

[Kazam_screencast_00018.webm](https://github.com/status-im/status-desktop/assets/20650004/491422b2-ae86-4063-bfb6-5694979cc058)

### CODE REVIEW

**Comment by caybro** [comment]
2 probs I see from the video:

image

- check the title (undefined?)
- it mentions "How many of the ... 67 remaining" would you like to burn but the remaining amount is not 67

**Review by caybro — APPROVED** [review]
LGTM!

**Review by alexjba — COMMENTED** [review]
The code changes look good to me!

I was able to test it only on storybook. There is one new issue I couldn't see on `master`.
Storybook is crashing if the BurnTokensPopup is resized. Probably we won't see it in the app because we enforce minimum width/height. @micieslak Can you also reproduce it on linux?

Looks like an infinite loop and the console is flooded with these warnings:
```
Qt Quick Layouts: Polish loop detected. Aborting after two iterations.
```

**Comment by micieslak** [comment]
> 

Everything fine in my env, no warning, errors, crashes. I'm on Ubuntu, Qt 5.15.2

**Comment by micieslak** [comment]
> The code changes look good to me!
> 
> I was able to test it only on storybook. There is one new issue I couldn't see on `master`. Storybook is crashing if the BurnTokensPopup is resized. Probably we won't see it in the app because we enforce minimum width/height. @micieslak Can you also reproduce it on linux?
> 
> Looks like an infinite loop and the console is flooded with these warnings:
> 
> ```
> Qt Quick Layouts: Polish loop detected. Aborting after two iterations.
> ```

Everything fine in my env, no warning, errors, crashes. I'm on Ubuntu, Qt 5.15.2

**Comment by caybro** [comment]
> > The code changes look good to me!
> > I was able to test it only on storybook. There is one new issue I couldn't see on `master`. Storybook is crashing if the BurnTokensPopup is resized. Probably we won't see it in the app because we enforce minimum width/height. @micieslak Can you also reproduce it on linux?
> > Looks like an infinite loop and the console is flooded with these warnings:
> > ```
> > Qt Quick Layouts: Polish loop detected. Aborting after two iterations.
> > ```
> 
> Everything fine in my env, no warning, errors, crashes. I'm on Ubuntu, Qt 5.15.2

Different Qt versions most likely

**Comment by alexjba** [comment]
Will look on it to see if I can spot the issue.

File: ui/app/AppLayouts/Communities/popups/BurnTokensPopup.qml [hunk 1]
```diff
@@ -99,53 +107,41 @@ StatusDialog {
...
+
+                    maximumExceededErrorText: qsTr("Exceeds available remaining")
                 }
             }
 
             StatusRadioButton {
                 id: allTokensButton
 
                 Layout.alignment: Qt.AlignTop
+                Layout.fillWidth: true
```
Comment (alexjba) [inline ui/app/AppLayouts/Communities/popups/BurnTokensPopup.qml:144]: @micieslak WDYT of setting an explicit width for the popup? This way the binding loop will be fixed and we'll make sure it won't crash.

This `StatusRadioButton` is misbehaving and triggers the polish loop and crash.

When the text inside this Control is long enough and the popup is resized, the text line count will increase/decrease, triggering an update in the `ColumnLayout` height and Popup height. Somehow the combination of Control height change + Layout and Popup height change triggers a polish loop. But not always. It's not 10/10 repro. Sometimes the width change works fine.

Either way, we won't see this crash in the app as long as the app minimum width is larger than 600px.

**Review by alexjba — APPROVED** [review]

File: ui/app/AppLayouts/Communities/popups/BurnTokensPopup.qml (hunk 1 above)
Comment (micieslak) [reply ui/app/AppLayouts/Communities/popups/BurnTokensPopup.qml]: Thanks @alexjba for your input. Fixed by reducing number of layouts and setting explicit width taken from the design as you suggested.

**Pull request merged**